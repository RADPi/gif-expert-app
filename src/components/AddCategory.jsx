import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState('')

	const onInputChange = ({ target }) => {
		setInputValue(target.value)
	}

	const onSubmit = e => {
		e.preventDefault()
		if (inputValue.trim().length < 2) return
		onNewCategory(inputValue.trim())
		setInputValue('')
	}

	return (
		<form aria-label='form' onSubmit={onSubmit}>
			<input
				type='text'
				placeholder='Buscar gifs'
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	)
}

AddCategory.propTypes = {
	onNewCategory: PropTypes.func.isRequired,
}
