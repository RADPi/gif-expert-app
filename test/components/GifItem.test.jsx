import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en <GifItem />', () => {
	const title = 'Titulo de Prueba'
	const url = 'http://www.url.com/prueba.jpg'

	test('Debe coincidir el snapshot', () => {
		const { container } = render(<GifItem title={title} url={url} />)
		expect(container).toMatchSnapshot()
	})

	test('Debe mostrar la imagen con la url y el alt indicado', () => {
		render(<GifItem title={title} url={url} />)
		const { src, alt } = screen.getByRole('img')
		expect(src).toBe(url)
		expect(alt).toBe(title)
	})

	test('Debe mostrar el titulo en el componente', () => {
		render(<GifItem title={title} url={url} />)
		expect(screen.getByText(title)).toBeTruthy()
	})
})
