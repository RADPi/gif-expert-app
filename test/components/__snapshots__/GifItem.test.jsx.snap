// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe coincidir el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de Prueba"
      src="http://www.url.com/prueba.jpg"
    />
    <p>
      Titulo de Prueba
    </p>
  </div>
</div>
`;
